xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.h

```diff
@@ -166,18 +168,42 @@ struct CVaapiProcessedPicture
   bool crop;
 };
 
-class CVaapiRenderPicture : public CVideoBuffer
+class CVaapiRenderPicture : public IVideoBufferDRMPRIME
 {
 public:
-  explicit CVaapiRenderPicture(int id) : CVideoBuffer(id) { }
+  explicit CVaapiRenderPicture(int id) : IVideoBufferDRMPRIME(id) { }
+  ~CVaapiRenderPicture() = default;
   void GetPlanes(uint8_t*(&planes)[YuvImage::MAX_PLANES]) override;
   void GetStrides(int(&strides)[YuvImage::MAX_PLANES]) override;
+
+  // IVideoBufferDRMPRIME
+#if VA_CHECK_VERSION(1, 1, 0)
+  AVDRMFrameDescriptor* GetDescriptor() const override;
+  uint32_t GetWidth() const override { return DVDPic.iWidth; }
+  uint32_t GetHeight() const override { return DVDPic.iHeight; }
+  int GetColorEncoding() const override;
+  int GetColorRange() const override;
+
+  bool Map() override;
+  void Unmap() override;
+#else
+  AVDRMFrameDescriptor* GetDescriptor() const override { return nullptr; }
+  uint32_t GetWidth() const override { return DVDPic.iWidth; }
+  uint32_t GetHeight() const override { return DVDPic.iHeight; }
```

Comment:
Since `GetWidth`/`GetHeight` is using the same implementation in both `#if`/`#else` blocks maybe it should be moved outside the blocks?